demo_iphour.py

```diff
@@ -0,0 +1,294 @@
+            time.sleep(20)
+
+            # We have to find the possible errors that can happen.
+            
+def thread_get_weather_info(tokenOWM=api_OpenWeather_token, cityid=api_OpenWeather_yourCity):
+    '''
+    get the weather info for my city
+    '''
+    global disp_string_weatherInfo
+    while True:
+        try:
+            base_url='https://api.openweathermap.org/data/2.5/weather?q={}&units=metric&appid={}'
+            api_OpenWeather_request=requests.get(base_url.format(cityid, tokenOWM))
+            api_OpenWeather_json=api_OpenWeather_request.json()
+            disp_string_weatherInfo=str(round(api_OpenWeather_json['main']['temp'])) + "￟C - " + api_OpenWeather_json['weather'][0]['description'] + " - " + api_OpenWeather_json['name']
+            print(str(datetime.now()) + " " + "thr4_weatherinfo got an update: " + disp_string_weatherInfo)
+            time.sleep(600)
+        except KeyError:
+            disp_string_weatherInfo="JSON Key error on weather info response. Check log. Retrying in 5 min."
+            print(str(datetime.now()) + " thr4_weatherinfo got an API error. \n" + str(api_OpenWeather_json))
+            time.sleep(360)
+        except ConnectionError:
+            disp_string_weatherInfo="Connection Error while getting the weather info. Will try again in 10 seconds."
+            print(str(datetime.now()) + " thr4_weatherinfo got a ConnectionError. will try again in 10 seconds.")
+            time.sleep(10)
+        except ValueError:
+            disp_string_weatherInfo="JSON Decode Error while getting the weather info. Will try again in 20 seconds."
+            print(str(datetime.now()) + " thr4_weatherinfo got a ValueError. will try again in 20 seconds.")
+            time.sleep(20)
+
+# Taken from the raspberry pi guy's sample, demo_scrollingtext.py
+def long_string(display, text='', num_line=2, num_cols=16, speed=0.1):
+    """ 
+    Parameters: (driver, string to print, number of line to print, number of columns of your display)
+    Return: This function send to display your scrolling string.
+    """
+    if len(text) > num_cols:
+        display.lcd_display_string(text[:num_cols], num_line)
+        time.sleep(3)
+        for i in range(len(text) - num_cols + 1):
+            text_to_print = text[i:i+num_cols]
+            display.lcd_display_string(text_to_print, num_line)
+            time.sleep(speed)
+        time.sleep(1)
+    else:
+        display.lcd_display_string(text, num_line)
+
+def get_ip():
+    '''function to get the ip of this machine
+       source: https://stackoverflow.com/questions/166506/finding-local-ip-addresses-using-pythons-stdlib#answer-28950776'''
+    s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    try:
+        # doesn't even have to be reachable
+        s.connect(('10.255.255.255', 1))
+        IP = s.getsockname()[0]
+    except:
+        IP = '127.0.0.1'
+    finally:
+        s.close()
+    return IP
```

Comment:
in GNU/Linux systems, such as Raspbian OS, you can use `hostname` to get such info directly from the host (check `hostname --help` for info and options).  no need to use sockets for that, just import check_output from subprocess (`from subprocess import check_output`), call `hostname -I` via check_output and parse it (`check_output(['hostname', '-I']).split()[0]` should get the local IPv4 address).

this is just an observation, not a required change.

Reply:
Hi @cgomesu I made all changes, just not this one, but I will definitely try it later to see how it works.

Reply:
I can already see the changes in the commit below (b9d604840db023ae81d463ace8345bc069a1eb75).  no need to do anything else right now.  I will review the changes you made later today and will let you know if there's anything else to be done before merging.